### PR TITLE
Add filter to facilities depending on functional/non-functional

### PR DESCRIPTION
**Pull request by WryanCartie**
Created the filter functionality in the frontend for facilities.

### CODE REVIEW

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 1]
```diff
@@ -7,14 +7,27 @@ $(function () {
     addFacility_handler();
     //handle delete facility
     deleteFacility_handler();
+    //handle change in filters in facility
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:10]: Why is there a comment? If the code requires comments that means it's not clear enough and it's the code that needs to be changed

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 2]
```diff
@@ -7,14 +7,27 @@ $(function () {
     addFacility_handler();
     //handle delete facility
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:8]: Remove this comment too, I suggest refactoring this entire string of function calls. Functions should always be verb phrases. These are nouns

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 3]
```diff
@@ -7,14 +7,27 @@ $(function () {
...
     deleteFacility_handler();
+    //handle change in filters in facility
+    facilityFilter_handler();
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:11]: Function should be verb phrases

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 4]
```diff
@@ -7,14 +7,27 @@ $(function () {
...
 });
 
 function getFacilitiesInfo() {
+
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:15]: [LINT] Unnecessary newline

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 5]
```diff
@@ -7,14 +7,27 @@ $(function () {
...
         if (res != null) {
+            $("#facilityList").empty();
+            let functionalFacilities = $('input[name=functionalFacilitiesDisplay]:checked').val();  
+
+        
+            if (functionalFacilities === 'functional') {
+                res = res.filter(facility=>facility.isFunctional)
+            } else if (functionalFacilities === 'non-functional') {
+                res = res.filter(facility=>!facility.isFunctional)
+            }
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Replace all magic strings here with constants. Avoid magic strings and numbers

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 3 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:11]: Alright

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 6]
```diff
@@ -228,3 +241,11 @@ function deleteFacility_handler() {
         })
     })
 }
+
+function facilityFilter_handler(){
+    $('input[type=radio][name=functionalFacilitiesDisplay]').on('change', e => {
+        e.preventDefault();
+        getFacilitiesInfo()
+    });
+
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:250]: [LINT] Unnecessary newline

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 7]
```diff
@@ -228,3 +241,11 @@ function deleteFacility_handler() {
         })
     })
 }
+
+function facilityFilter_handler(){
+    $('input[type=radio][name=functionalFacilitiesDisplay]').on('change', e => {
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:246]: Same here, no magic strings

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:246]: Wait, is this HTML?

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 5 above)
Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Are these HTML?

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 1 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:10]: I just follow the predecessor usage of comments there

File: src/views/region/facilitiesInfo.html [hunk 8]
```diff
@@ -13,6 +13,26 @@
...
+                        </div>
+                        <div class="form-check form-check-inline">
+                            <input type="radio" id="rbFacilitiesNonFunctional" name="functionalFacilitiesDisplay" value="non-functional" class="form-check-input" >
+                            <label for="rbFacilitiesNonFunctional" class="form-check-label">Non Functional</label>
+                        </div>
+                    </div>
+                </form>
+                  
+          
+
```
Comment (Phobez) [inline src/views/region/facilitiesInfo.html:35]: Too many empty newlines

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 1 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:10]: I will remove it

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 2 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:8]: Alright

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 3 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:11]: I will fix that

**Review by Phobez — CHANGES_REQUESTED** [review]
As per comments

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 4 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:15]: Sure lemme fix

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 7 above)
Comment (extremebip) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:246]: Yes, that's jQuery selector string.

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 5 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: What magic string?

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 6 above)
Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:250]: Sure

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 5 above)
Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Magic strings are strings directly in code. There's a lot of it here. Extract them into constant variables if possible, but I'm asking if this is HTML?

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 7 above)
Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:246]: Hm, not sure what the best practice is for HTML strings (whether they should be extracted). Will find out later, for now, resolved.

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 1 above)
Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:10]: > I just follow the predecessor usage of comments there

Only do this if the code is very complex. If it is changeable, then change it. `Leave code better than when you found it`

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js (hunk 5 above)
Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: From discussion on Discord: `functional` and `non-functional` are not HTML so they should be extracted. I suggest an `enum` (do we have that in JS?)

Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: I think the typical thing to do is store them as constants like const FUNCTIONAL = 'functional', we can do an enum file but thats way more complex

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Can't we put the enum in the same file?

Comment (extremebip) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: We do have enum in JS, but we also need to put the enum in the HTML.

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Hm, then just `const` is fine

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 9]
```diff
@@ -1,28 +1,42 @@
...
-    //handler for facility  add
-    addFacility_handler();
-    //handle delete facility
-    deleteFacility_handler();
+
+    openCloseAllFacilities();
+
+    addFacility();
+
+    deleteFacility();
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:12]: Aren't these handlers too? Shouldn't they be `add listener` too?

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 10]
```diff
@@ -1,28 +1,42 @@
...
     ipcRenderer.send('Facility:getFacilitiesByRegion', parseInt(window.process.argv.slice(-1)));
     ipcRenderer.once('Facility:getFacilitiesByRegionOK', (e, res) => {
         dataAvailable = true;
         if (res != null) {
+            $("#facilityList").empty();
+            let functionalFacilities = $('input[name=functionalFacilitiesDisplay]:checked').val();
+
+            if (functionalFacilities === FUNCTIONAL) {
+                res = res.filter(facility => facility.isFunctional)
+            } else if (functionalFacilities === NON_FUNCTIONAL) {
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: How can `functionalFacilities` be `NON_FUNCTIONAL`? What does `$('input[name=functionalFacilitiesDisplay]:checked').val();` actually do?

Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: It basically checks the value of a group of radio buttons for what value it is. I admit the functionalFacilities is also followign the ones in components where it is usedComponents. I guess i should change he name to something clearer, like filterInput

Comment (Phobez) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Ah, I think `facilityFilter is good, think `if (facilityFilter === FUNCTIONAL)`

Comment (WryanCartie) [reply src/renderers/PartialViewRenderers/region/facilitiesInfo.js:29]: Alright Fixed.

File: src/renderers/PartialViewRenderers/region/facilitiesInfo.js [hunk 11]
```diff
@@ -1,28 +1,42 @@
...
-    //handler for facility  add
-    addFacility_handler();
-    //handle delete facility
-    deleteFacility_handler();
+
+    addOpenCloseListListener();
+
+    addNewFacilityListener();
+
+    AddDeleteListener();
```
Comment (Phobez) [inline src/renderers/PartialViewRenderers/region/facilitiesInfo.js:12]: Why is this capitalised?

**Review by Phobez — APPROVED** [review]

**Pull request merged**